lib/spack/spack/test/cmd/diff.py

```diff
@@ -60,17 +61,21 @@ def test_load_first(install_mockery, mock_fetch, mock_archive, mock_packages):
     assert "intersect" in result and len(result['intersect']) > 50
 
     # After we install another version, it should ask us to disambiguate
-    install('mpileaks+debug')
+    install_cmd('mpileaks+debug')
 
     # There are two versions of mpileaks
     with pytest.raises(spack.main.SpackCommandError):
-        diff('mpileaks', 'mpileaks+debug')
+        diff_cmd('mpileaks', 'mpileaks+debug')
 
     # But if we tell it to use the first, it won't try to disambiguate
-    assert "variant" in diff('--first', 'mpileaks', 'mpileaks+debug')
+    assert "variant" in diff_cmd('--first', 'mpileaks', 'mpileaks+debug')
 
     # This matches them exactly
-    output = diff("--json", "mpileaks@2.3/ysubb76", "mpileaks@2.3/ft5qff3")
+    hashes = find_cmd('--format', '{hash}', 'mpileaks@2.3')
+    hash_list = hashes.split()
+    output = diff_cmd("--json",
+                      "mpileaks@2.3/{0}".format(hash_list[0]),
+                      "mpileaks@2.3/{0}".format(hash_list[1]))
```

Comment:
You need to do something here to ensure that one of these is the debug spec, beyond relying on the order to always stay the same.

```suggestion
    debug_hash = find_cmd('--format', '{hash}', 'mpileaks+debug').strip()
    no_debug_hashes = find_cmd('--format', '{hash}', 'mpileaks~debug')
    no_debug_hash = no_debug_hashes.split()[0]
    output = diff_cmd("--json",
                      "mpileaks/{0}".format(debug_hash),
                      "mpileaks/{0}".format(no_debug_hash))
```